src/sentry/integrations/notion/client.py

```diff
@@ -0,0 +1,106 @@
+from django.utils.encoding import force_bytes
+
+from sentry import options
+from sentry_plugins.client import ApiClient
+
+DATABASES = "/search"
+PAGES = "/pages"
+GET_PAGE = "/pages/%s"
+APPEND_BLOCK = "/blocks/%s/children"
+OAUTH = "/oauth/token"
+
+NOTION_VERSION = "2021-05-13"
+NOTION_API_URL = "https://api.notion.com/v1"
+
+
+class NotionApiClient(ApiClient):
+    base_url = NOTION_API_URL
+    plugin_name = "notion"
+
+    def __init__(self, api_key=None, **kwargs):
+        self.api_key = api_key
+        super().__init__(**kwargs)
+
+    def request(self, method="GET", path="", data=None, params=None, **kwargs):
+        params = {} if params is None else params.copy()
+        headers = {}
+        headers["Authorization"] = "Bearer " + self.api_key
+        headers["Notion-Version"] = NOTION_VERSION
+        return self._request(method, path, data=data, params=params, headers=headers, **kwargs)
+
+    def authenticate(self, code, redirect_uri, **kwargs):
+        from base64 import b64encode
+
+        client_id = options.get("notion-app.client-id")
+        client_secret = options.get("notion-app.client-secret")
+        encoded_auth_string = b"Basic " + b64encode(force_bytes(client_id + ":" + client_secret))
+
+        headers = {}
+        headers["Authorization"] = encoded_auth_string
+        headers["Notion-Version"] = NOTION_VERSION
+        data = {"grant_type": "authorization_code", "code": code, "redirect_uri": redirect_uri}
+
+        return self._request(method="POST", path=OAUTH, data=data, headers=headers, **kwargs)
+
+    def get_databases(self):
+        """
+        Query databases in Notion
+        """
+        data = {"filter": {"value": "database", "property": "object"}}
+        return self.request(path=DATABASES, method="POST", data=data)
+
+    def get_databases_list(self):
+        """
+        Return databases options to use in a Django form
+        """
+        databases = self.get_databases()
+        return [(db["id"], db["title"][0]["plain_text"]) for db in databases["results"]]
```

Comment:
Feel like this should be wrapped in try/catch, if `db["title"]` has no elements -> It will error